genesyscloud/idp_salesforce/genesyscloud_idp_salesforce_proxy.go

```diff
@@ -0,0 +1,92 @@
+package idp_salesforce
+
+import (
+	"context"
+	"github.com/mypurecloud/platform-client-sdk-go/v129/platformclientv2"
+)
+
+/*
+The genesyscloud_idp_salesforce_proxy.go file contains the proxy structures and methods that interact
+with the Genesys Cloud SDK. We use composition here for each function on the proxy so individual functions can be stubbed
+out during testing.
+*/
+
+// internalProxy holds a proxy instance that can be used throughout the package
+var internalProxy *idpSalesforceProxy
+
+// Type definitions for each func on our proxy so we can easily mock them out later
+type getIdpSalesforceByIdFunc func(ctx context.Context, p *idpSalesforceProxy) (salesforce *platformclientv2.Salesforce, resp *platformclientv2.APIResponse, err error)
+type updateIdpSalesforceFunc func(ctx context.Context, p *idpSalesforceProxy, salesforce *platformclientv2.Salesforce) (*platformclientv2.Identityprovider, *platformclientv2.APIResponse, error)
+type deleteIdpSalesforceFunc func(ctx context.Context, p *idpSalesforceProxy) (resp *platformclientv2.APIResponse, err error)
+
+// idpSalesforceProxy contains all of the methods that call genesys cloud APIs.
+type idpSalesforceProxy struct {
+	clientConfig             *platformclientv2.Configuration
+	identityProviderApi      *platformclientv2.IdentityProviderApi
+	getIdpSalesforceByIdAttr getIdpSalesforceByIdFunc
+	updateIdpSalesforceAttr  updateIdpSalesforceFunc
+	deleteIdpSalesforceAttr  deleteIdpSalesforceFunc
+}
+
+// newIdpSalesforceProxy initializes the idp salesforce proxy with all of the data needed to communicate with Genesys Cloud
+func newIdpSalesforceProxy(clientConfig *platformclientv2.Configuration) *idpSalesforceProxy {
```

Comment:
No getAll functionality

Reply:
My understanding of this resource is that there is no get all. If a salesforce object exists there is only one of them so the exporter just checks if that object exists or not